crates/trigger-http/src/handler.rs

```diff
@@ -188,31 +183,33 @@ impl HttpHandlerExecutor {
             Handler2023_10_18(IncomingHandler2023_10_18),
         }
 
-        let handler = match instance
-            .exports(&mut store)
-            .instance("wasi:http/incoming-handler@0.2.0-rc-2023-10-18")
-        {
-            Some(mut instance) => Some(Handler::Handler2023_10_18(IncomingHandler2023_10_18::new(
-                &mut instance,
-            )?)),
-            None => None,
-        };
-        let handler = match handler {
-            Some(handler) => Some(handler),
-            None => match instance
-                .exports(&mut store)
-                .instance("wasi:http/incoming-handler@0.2.0-rc-2023-11-10")
+        let handler =
             {
-                Some(mut instance) => Some(Handler::Handler2023_11_10(
-                    IncomingHandler2023_11_10::new(&mut instance)?,
-                )),
-                None => None,
-            },
-        };
-        let handler = match handler {
-            Some(handler) => handler,
-            None => Handler::Latest(Proxy::new(&mut store, &instance)?),
-        };
+                let mut exports = instance.exports(&mut store);
+                match ty {
+                    HandlerType::Wasi2023_10_18 => {
+                        let mut instance = exports
+                            .instance(WASI_HTTP_EXPORT_2023_10_18)
+                            .ok_or_else(|| {
+                                anyhow!("export of `{WASI_HTTP_EXPORT_2023_10_18}` not an instance")
+                            })?;
+                        Handler::Handler2023_10_18(IncomingHandler2023_10_18::new(&mut instance)?)
+                    }
+                    HandlerType::Wasi2023_11_10 => {
+                        let mut instance = exports
+                            .instance(WASI_HTTP_EXPORT_2023_11_10)
+                            .ok_or_else(|| {
+                                anyhow!("export of `{WASI_HTTP_EXPORT_2023_11_10}` not an instance")
+                            })?;
+                        Handler::Handler2023_11_10(IncomingHandler2023_11_10::new(&mut instance)?)
+                    }
+                    HandlerType::Wasi0_2 => {
+                        drop(exports);
+                        Handler::Latest(Proxy::new(&mut store, &instance)?)
+                    }
+                    HandlerType::Spin => unreachable!(),
```

Comment:
nit: I prefer using a `panic` with a message just in case (through some refactoring) this becomes technically reachable.

Reply:
`panic!("unreachable")`